source/features/conversation-navigation.tsx

```diff
@@ -0,0 +1,258 @@
+const ITEMS_PER_PAGE = 25;
+
+type Conversation = {
+	cursor: string;
+	number: number;
+	url: string;
+};
+
+const fetchConversationList = cache.function(
+	async ({query, page}: ReturnType<typeof getListQuery>): Promise<Conversation[]> => {
+		const {after} = getPageCursors(page);
+
+		/**
+		 * When fetching a page of conversations we want to also fetch last item from previous page, and first item from next page.
+		 * This is needed for cases when we are on a conversation that is last in current query page, and we need to know if items exist on next page.
+		 * To take into consideration also, than when we are on first item on first page, we don't have a previous item, and we don't want to fetch using a negative cursor.
+		 * Read more on {@link getCursor} function
+		 */
+
+		const previousAfter = page === 1 ? after : getPreviousCursor(after);
+		const fetchItemsCount = page === 1 ? ITEMS_PER_PAGE + 1 : ITEMS_PER_PAGE + 2;
+		const searchQuery = getConversationSearchQuery(fetchItemsCount, previousAfter, query);
+		const {search} = await api.v4(searchQuery);
+
+		return search.edges.map((edge: any) => ({
+			...edge.node,
+			cursor: edge.cursor
+		}));
+	},
+	{
+		cacheKey: ([listQuery]) => `${__filebasename}:${JSON.stringify(listQuery)}`,
+		maxAge: 1 / 24 // 1 hour from a day is enough
+	}
+);
+
+async function getConversationList() {
+	const listQuery = getListQuery();
+	const list = await fetchConversationList(listQuery);
+	const currentNumber = getConversationNumber();
+	const currentConversation = list.find(item => item.number === currentNumber);
+	const {after, before} = getPageCursors(listQuery.page);
+
+	// `currentConversation` is from previous page, we need to fetch previous page items
+	if (currentConversation?.cursor === after) {
+		const newListQuery = {
+			...listQuery,
+			page: listQuery.page - 1
+		};
+		return {
+			list: await fetchConversationList(newListQuery),
+			listQuery: newListQuery
+		};
+	}
+
+	// `currentConversation` is from next page, we need to fetch next page items
+	if (currentConversation?.cursor === before) {
+		const newListQuery = {
+			...listQuery,
+			page: listQuery.page + 1
+		};
+		return {
+			list: await fetchConversationList(newListQuery),
+			listQuery: newListQuery
+		};
+	}
+
+	return {
+		list,
+		listQuery
+	};
+}
+
+function noListQuery(): boolean {
+	return !(
+		// Allowed also from global conversation list
+		pageDetect.isConversationList(new URL(document.referrer)) ||
+		(new URL(location.href)).searchParams.has('q')
+	);
+}
+
+function getListQuery() {
+	const referrerUrl = new URL(document.referrer);
+
+	// Coming from another issue with query in URL
+	if (!pageDetect.isConversationList(referrerUrl)) {
+		return parseConversationListURL(new URL(location.href));
+	}
+
+	// Default conversation lists, even global, don't have query in URL
+	// But it seems there is some refined-github feature that adds a default query in URL based on query input
```

Comment:
GitHub does have default queries, for example `/pulls` implies `is:open is:pr author:@me archived:false`. This is by default.

You'll be happy to know that this issue is already handled by `search-query.ts`

https://github.com/sindresorhus/refined-github/blob/2863dddde6fa72d9dd85fea242e69ed650b6cf88/source/features/sort-conversations-by-update-time.tsx#L23

This means that `getListQuery` could be replaced with

```ts
function getConversationList(): URL | location {
	const referrerUrl = new URL(document.referrer);
	return pageDetect.isConversationList(referrerUrl) ? referrerURL : location;
}
```

And then in `getConversationList`:

```diff
  function getConversationList() {
-	const listQuery = getListQuery();
+   const listUrl = getConversationList();
+   const query = new SearchQuery(listUrl).get();
+   const page = listUrl.searchParams.get('page') ?? '1';
```